### DISCUSSION TITLE
Unable to Add Accessory - The Setup code is incorrect

### DISCUSSION THREAD

**Discussion by Dopamine404** [HomeSpan/HomeSpan#814] [Q&A]
I successfully implemented an ESP32 using the HomeSpan library, following a YouTube tutorial, and managed to connect without any errors initially. However, after a few days, I encountered an issue where the Apple Home app displays an error message stating **_"Unable to add accessory: The setup code is incorrect."_**

Serial Monitor displayed after entering setup code "*** ERROR: SRP Proof Verification Failed"

**Steps Taken:**

Changed ESP32 devices to rule out hardware issues.
Tried different Apple devices to ensure the problem was not device-specific.
Cleared all WiFi data and cache on the ESP32.
Reset the WiFi router to eliminate any potential network conflicts.
Issue Persisting:

Despite the above efforts, the problem persists, and I'm unable to resolve it. I'm seeking guidance on potential solutions or further troubleshooting steps to address this issue and successfully connect to the Home app again.

Additional Information:

The initial setup was completed without any errors following a YouTube tutorial.
No changes were made to the setup prior to encountering this issue.
Any insights, suggestions, or alternative methods to resolve this issue would be greatly appreciated.

**Comment by HomeSpan** [comment]
I assume above that you've been un-pairing and then re-pairing your device, and this is when you've experienced the error.  In other words, you did NOT have a device that was already paired and working properly suddenly cause this error (the error can only occur when you try to pair, not once the device is already paired).

Assuming you are getting this error after trying to pair an unpaired device, how are you defining the Setup Code?  In the CLI using the 'S' command? Or in the sketch using `homeSpan.setPairingCode()`?

Also, how are you inputting the setup code into the Home App?  Typing it directly?

Finally, if you have entered a setup code from the CLI using the 'S' command, make sure NOT to erase the flash when uploading a new sketch, since this erases what you stored and the fall back is for HomeSpan to use the default code 46637736.

**Comment by Dopamine404** [comment]
I'm using `homeSpan.setPairingCode()` and input the setup code directly by typing.

**Reply by HomeSpan** [reply]
Ah, in that case see #786 for issue and resolution.

**Comment by Dopamine404** [comment]
I re-enter the setup code using CLI and after it connected successfully. After that, I reprogrammed ESP32 and tried to connect with the app that time connected without any error.
